File: hw1/pow_full.rb

def powI(pow, base)
	acc = 1
	p = 0
	for p in (0..pow - 1)
		acc = acc * base
	end
	return acc
end

def powF(pow, base)
	return 1 if pow == 0
	return base if pow == 1
	return powF(pow - 1, base) * base
end

if ARGV.length != 2 
  	puts "fact.rb usage: [NUMBER]" 
  	exit
end

puts powI(ARGV[0].to_i, ARGV[1].to_i)
puts powF(ARGV[0].to_i, ARGV[1].to_i)

File: hw1/pow_full.py
#! /usr/bin/env python
import sys

def powI(pow1, base):
	acc = 1
	p = 0
 	for p in range(0, pow1):
   		acc = acc * base
  	
  	return acc

def powF(pow1, base):
	if (pow1 == 0):
		return 1
	
	elif (pow1 == 1):
		return base
	
	else:
		return powF(pow1 - 1, base) * base
	

if len(sys.argv) != 3:
  	print("%s usage: [NUMBER]" % sys.argv[0])
  	exit()

arg1 = int(sys.argv[1])
arg2 = int(sys.argv[2])
print(powF(arg1, arg2))
	
  	
  
